IPs/RSKIP176.md

```diff
@@ -0,0 +1,115 @@
+
+## Motivation
+
+Allowing users to transfer BTC to RSK immediately, without waiting for usual peg-in confirmations. This also generates a market between user who want to fast peg-in and those who count on liquidity.
+
+## Specification
+
+The fast-bridge system connects users, Liquidity Providers (LP) and a certain Liquidity Bridge Contract (LBC).
+This RSKIP specifies the requirements for consensus changes without limiting the actual implementation of the fast-bridge system.
+
+#### Bridge
+
+The bridge needs to expose a new method called **registerBtcTransfer**().
+
+#### registerBtcTransfer()
+
+**ABI Signature**
+
+        function registerBtcTransfer(bytes btcTxSerialized, int heigth, bytes pmtSerialized, bytes32 derivationArgumentsHash, string userRefundBtcAddress, string LiquidityBridgeContractAddress, string LiquidityProviderBtcAddress, bool shouldTransferToContract) return int
+
+
+**Derivation Arguments**
+
+The derivation arguments are values used to close the agreement between the Liquidity Provider and the user. Some of these parameters are the RSK address where funds will be received, the BTC refund address, the Liquidity Bridge Contract address, the value to transfer, the amount of BTC blocks that the Liquidity Provider should wait before advancing the funds, the Liquidity Provider BTC refund address, the derived Federation address, and some other values that are domain of the Liquidity Provider.
+
+
+**Parameters**:
+- *byte[]* btcTxSerialized: Serialized Bitcoin transaction following Bitcoin serialization
+- *int* height: BTC block number where the transaction is present
+- *byte[]* pmtSerialized: Serialized partial merkle tree (following bitcoinj structure) that proves that the transaction belongs to the indicated block
+- *Sha256Hash* derivationArgumentsHash: A 32 bytes hash created from the derivation arguments
+- *string* userRefundBtcAddress: Base58 string representing the user BTC refund address
+- *string* LiquidityBridgeContractAddress: String representing Liquidity Bridge Contract address
+- *string* LiquidityProviderBtcAddress: Base58 string representing  the Liquidity Provider BTC refund address
+- *bool* shouldTransferToContract: Status provided by Liquidity Provider Contract to know if should transfer value to contract or not
+
+
+**Flow**:
+
+- Bridge receives a call from a contract (an alleged Liquidity Bridge Contract) and validates that the sender of that call is effectively the LBC address, that is received as parameter
+
+- Performs the same validations the Bridge does in **registerBtcTransaction** to determine if the BTC transaction is legitimate (e.g.: txAlreadyProcessed, confirmations, PMT, etc)
+
+- Verifies tx is sending funds to address derivated from Federation using the provided derivation arguments and hashing them:
+
+	`Sha256Hash(derivationArgumentsHash, userRefundAddress, LBCAddress, LPBtcAddress)`
+
+- Checks if the derivation hash was used already. If so, it will refund the user.
```

Comment:
According to what we discussed, entries saved on storage will stay there for a week. After so, the entry will be erased.

Reply:
It will be stored until the UTXO is spent. It could be by migration, by consuming the UTXO or because it is spent by a consolidation process after N days.